### PR TITLE
issue-70: pass in parameters for preconditions during planning

### PR DESCRIPTION
**Pull request by scottgunther**
Resolves issue: #70

Changes proposed in this pull request:
- Pass in parameters for preconditions during planning
- Update old forward state space search with some newer changes

@GannettDigital/simulato-core-contributors

### CODE REVIEW

File: lib/planner/search-algorithms/forward-state-space-search-heuristic.js [hunk 1]
```diff
@@ -147,14 +147,22 @@ module.exports = forwardStateSpaceSearch = {
     },
     * _checkPreconditions(node, component, actionName, callback) {
         let next = yield;
+        let action = component.actions[actionName];
 
         if (component.actions[actionName].preconditions === undefined) {
```
Comment (tomrdale) [inline lib/planner/search-algorithms/forward-state-space-search-heuristic.js]: can just use action here now that you are declaring it

File: lib/planner/search-algorithms/forward-state-space-search.js [hunk 2]
```diff
@@ -154,20 +153,28 @@ module.exports = forwardStateSpaceSearch = {
 
         callback();
     },
-    * _checkPreconditions(node, action, actionIdentifier, callback) {
+    * _checkPreconditions(node, component, actionName, callback) {
         let next = yield;
+        let action = component.actions[actionName];
 
-        if (action.preconditions === undefined) {
+        if (component.actions[actionName].preconditions === undefined) {
```
Comment (tomrdale) [inline lib/planner/search-algorithms/forward-state-space-search.js]: can just use action

File: lib/planner/search-algorithms/forward-state-space-search-heuristic.js (hunk 1 above)
Comment (scottgunther) [reply lib/planner/search-algorithms/forward-state-space-search-heuristic.js]: Great point. Thanks, I'll fix it now!

File: lib/planner/search-algorithms/forward-state-space-search.js (hunk 2 above)
Comment (scottgunther) [reply lib/planner/search-algorithms/forward-state-space-search.js]: Great point. Thanks, I'll fix it now!

**Pull request merged**